File: 3875-construct-uniform-parity-array-i/3875-construct-uniform-parity-array-i.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        int n = nums1.length;
        int[] nums2 = new int[n];
        int j = 0 , t = 0 , f = 0;
        for(int i = 0 ; i < n - 1 ; i++){
            nums2[i] = nums1[i] - nums1[j];
            j++;
        }
        for(int i = 0 ; i < n - 1 ; i++){
            if(nums2[i] % 2 == 0){
                f++;
            }
            else{
                t++;
            }
        }
        return f == n - 1 || t == n - 1;
    }
}

File: 3875-construct-uniform-parity-array-i/3875-construct-uniform-parity-array-i.py
class Solution(object):
    def uniformArray(self, nums1):
        n = len(nums1)
        l1 = []
        j , t , f = 0 , 0 , 0
        for i in range(0,n-1):
            l1.append(nums1[i] - nums1[j])
            j += 1
        for i in range(0,n-1):
            if l1[i] % 2 == 0:
                f += 1
            else:
                t += 1
        return t == n -1 or f == n - 1

        